dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ResourceLinkResource.java

```diff
@@ -0,0 +1,157 @@
+     * Given an inode or identifier this will build get you a Resource Link
+     * The inode nor identifier, is expected other wise you'll get exception
+     * @param request    {@link HttpServletRequest} http request
+     * @param response   {@link HttpServletResponse} http response
+     * @param inode      {@link String} asset inode
+     * @param identifier {@link String} identifier
+     * @param language   {@link String} optional parameter
+     * @return Response
+     * @throws DotDataException
+     * @throws DotStateException
+     * @throws DotSecurityException
+     */
+    @GET
+    @JSONP
+    @NoCache
+    @Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
+    public Response findResourceLink(@Context final HttpServletRequest  request,
+                                     @Context final HttpServletResponse response,
+                                     @QueryParam("inode")            final String inode,
+                                     @QueryParam("identifier")       final String identifier,
+                                     @DefaultValue("-1") @QueryParam("language") final String language) throws DotStateException {
+        try {
+            if (!UtilMethods.isSet(inode) && !UtilMethods.isSet(identifier)) {
+
+                throw new IllegalArgumentException("Missing required inode/identifier param");
+            }
+
+            final InitDataObject initData =
+                    new WebResource.InitBuilder(webResource).requiredBackendUser(true).requiredFrontendUser(false)
+                            .requestAndResponse(request, response).rejectWhenNoUser(true).init();
+            final User user       = initData.getUser();
+            final long languageId = LanguageUtil.getLanguageId(language);
+            final PageMode mode   = PageMode.get(request);
+            final Contentlet contentlet = this.getContentlet(inode, identifier, languageId,
+                    ()-> WebAPILocator.getLanguageWebAPI().getLanguage(request).getId(), initData, mode);
+
+            Logger.debug(this, ()-> "Finding the resource link for the contentlet: " + contentlet.getIdentifier());
+            final ResourceLink link     = new ResourceLink.ResourceLinkBuilder().build(request, user, contentlet);
+            if(link.isDownloadRestricted()) {
+
+                throw new DotSecurityException("The Resource link to the contentlet is restricted.");
+            }
+
+            return Response.ok(new ResponseEntityView(ImmutableMap.of("resourceLink",
+                    ImmutableMap.of(
+                            "href",        link.getResourceLinkAsString(),
+                            "text",        link.getResourceLinkUriAsString(),
+                            "mimeType",    link.getMimeType(),
+                            "idPath",      link.getIdPath(),
+                            "versionPath", link.getVersionPath()
+                    )
+            ))).build();
+        } catch (Exception ex) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5468320)